test/framework/src/main/java/org/elasticsearch/test/ExternalNode.java

```diff
@@ -112,7 +112,7 @@ synchronized void startInternal(Client client, Settings settings, String nodeNam
                 case "node.mode":
                 case "node.local":
                 case NetworkModule.TRANSPORT_TYPE_KEY:
-                case DiscoveryModule.DISCOVERY_TYPE_KEY:
+                case "discovery.type":
```

Comment:
case expressions unfortunately only allow string constants.

Java's interpretation of constant allows `NetworkModule.TRANSPORT_TYPE_KEY` which is a static final string, but `DiscoveryModule.DISCOVERY_TYPE_SETTING.getKey()` is unfortunately not a constant according to the Java language spec.